docs/src/pages/components/text-fields/FormPropsTextFields.js

```diff
@@ -0,0 +1,221 @@
+          label="Password"
+          className={classes.textField}
+          type="password"
+          autoComplete="current-password"
+          margin="normal"
+        />
+        <TextField
+          id="standard-read-only-input"
+          label="Read Only"
+          defaultValue="Hello World"
+          className={classes.textField}
+          margin="normal"
+          InputProps={{
+            readOnly: true,
+          }}
+        />
+        <TextField
+          id="standard-number"
+          label="Number"
+          type="number"
+          className={classes.textField}
+          InputLabelProps={{
+            shrink: true,
+          }}
+          margin="normal"
+        />
+        <TextField
+          id="standard-search"
+          label="Search field"
+          type="search"
+          className={classes.textField}
+          margin="normal"
+        />
+        <TextField
+          id="standard-helperText"
+          label="Helper text"
+          defaultValue="Default Value"
+          className={classes.textField}
+          helperText="Some important text"
+          margin="normal"
+        />
+      </div>
+      <div>
+        <TextField
+          required
+          id="filled-required"
+          label="Required"
+          defaultValue="Hello World"
+          className={classes.textField}
+          margin="normal"
+          variant="filled"
+        />
+        <TextField
+          disabled
+          id="filled-disabled"
+          label="Disabled"
+          defaultValue="Hello World"
+          className={classes.textField}
+          margin="normal"
+          variant="filled"
+        />
+        <TextField
+          id="filled-password-input"
+          label="Password"
+          className={classes.textField}
+          type="password"
+          autoComplete="current-password"
+          margin="normal"
+          variant="filled"
+        />
+        <TextField
+          id="filled-read-only-input"
+          label="Read Only"
+          defaultValue="Hello World"
+          className={classes.textField}
+          margin="normal"
+          InputProps={{
+            readOnly: true,
+          }}
+          variant="filled"
+        />
+        <TextField
+          id="filled-number"
+          label="Number"
+          type="number"
+          className={classes.textField}
+          InputLabelProps={{
+            shrink: true,
+          }}
+          margin="normal"
+          variant="filled"
+        />
+        <TextField
+          id="filled-search"
+          label="Search field"
+          type="search"
+          className={classes.textField}
+          margin="normal"
+          variant="filled"
+        />
+        <TextField
+          id="filled-helperText"
+          label="Helper text"
+          defaultValue="Default Value"
+          className={classes.textField}
+          helperText="Some important text"
+          margin="normal"
+          variant="filled"
```

Comment:
I've kept all variants due to the visual regression tests. cc @oliviertassinari 
Re-visiting where defaultValue is used sounds good. I guess we would also have to keep it for the disabled example too.
Adding a placeholder example sounds good.

Reply:
> I've kept all variants due to the visual regression tests.

Ah, right. Might be a good opportunity to move the bulk of them to `test/regressions`.

Reply:
> Only needs one variant. (Outlined?)

I was wondering about this one. I wondering what's the usage split ration between the 3 variants. We can get this data by looking at the interaction numbers:

![Capture d’écran 2019-10-13 à 17 29 08](https://user-images.githubusercontent.com/3165635/66717985-0cd0a100-eddf-11e9-8c99-c398ed231e03.png)

If I analyze it correctly: 58% standard, 33% outlined, 8% filled. 

Maybe we could have x3 more demos, one for each variant?

Reply:
Seems like a flawed analysis. They are view in descending quantity according to the order in which they are presented – proof that we don't need one demo per variant (other than the initial one to demonstrate the appearance of each variant).

Reply:
> Seems like a flawed analysis. They are view in descending quantity according to the order in which they are presented

This is very possible. Should we try to revert the order? I'm curious to see the results. Can we significantly influence what people use with the order? It would be great if we can, but it might take time to manifest in the data.

Reply:
No. Users are interested in the implementation, not the specific variant in question. (The implementation is the same regardless of the variant.)

1. Show the appearance of the three variants.
2. Show the implementation specifics with one variant.
3. User applies 1 and 2 together as required.

Reply:
> No. Users are interested in the implementation, not the specific variant in question.

Interesting, I haven't thought about this possibility! So if I understand the assumption correctly: if let say, we reverse the order of the *TextField* and *OutlinedTextField* demos, the % split will be more or less be the same because developers want to know the props to trigger, independently from the variant to apply?

Reply:
That's my interpretation, yes.

Reply:
Splitting all the demos in 3 would mean there are 14 demos? Doesn't sound right to me.

I would be happier with removing the implementation specifics from the other variants.

Reply:
For benchmark, there are 50 demos in https://getbootstrap.com/docs/4.3/components/forms/.

Reply:
> I would be happier with removing the implementation specifics from the other variants.

No object, at the end of the day, if we get no user that asks for a change, we might have done it right :). What do you think of creating these demos, but moving them to the visual  test folder?